src/main.go

```diff
@@ -75,14 +74,44 @@ func isTopicInList(topicList []string, searchedTopic string, username string, cl
 	replacer := strings.NewReplacer("%u", username, "%c", clientid)
 
 	for _, topicFromList := range topicList {
-		if common.TopicsMatch(replacer.Replace(topicFromList), searchedTopic) {
+		if topicsMatch(replacer.Replace(topicFromList), searchedTopic) {
 			return true
 		}
 	}
 	return false
 }
 
-func checkAccessToTopic(topic string, acc int, cache *userState, username string, clientid string) bool {
+func topicsMatch(savedTopic, givenTopic string) bool {
+	return givenTopic == savedTopic || match(strings.Split(savedTopic, "/"), strings.Split(givenTopic, "/"))
+}
+
+func match(route []string, topic []string) bool {
+	if len(route) == 0 {
+		if len(topic) == 0 {
+			return true
+		}
+		return false
+	}
+
+	if len(topic) == 0 {
+		if route[0] == "#" {
+			return true
+		}
+		return false
+	}
+
+	if route[0] == "#" {
+		return true
+	}
+
+	if (route[0] == "+") || (route[0] == topic[0]) {
+		return match(route[1:], topic[1:])
+	}
+
+	return false
+}
```

Comment:
Too bad that iegomez change the structur of his code. But yeah, to be safe for next builds an own implementation is better. Can you move that function in a separate file, to keep the code clean. And a few unit tests for some basic topics would be great.
Why did you use ifs instead of switch-case? 

https://github.com/iegomez/mosquitto-go-auth/blob/master/backends/topics/topics.go

Reply:
Sure, I can do that in the next update. 
Oh, I did not know it was moved to a separate file. I thought that part of code was removed so, I used the same code from the older versions which was written using if statements.